lib/Doctrine/ORM/Query/AST/Phase2OptimizedConditional.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query\AST;
+
+/**
+ * Marks types that can be used in place of a ConditionalExpression as a phase
+ * 2 optimization.
+ *
+ * @psalm-inheritors ConditionalPrimary|ConditionalFactor|ConditionalTerm
+ */
+interface Phase2OptimizedConditional
```

Comment:
I understand that it should be declared as internal because it exists just as an interim, right?.

Reply:
Not sure what you mean by iterim, but you're right, it should be marked as internal.

Reply:
Then I misunderstood the interface's lifetime.